src/magic_ter_fur_transform.cpp

```diff
@@ -0,0 +1,287 @@
+            JsonArray target_array = ter_obj.get_array( "valid_terrain" );
+
+            weighted_int_list<ter_id> results;
+            load_ter_results( ter_obj, "result", results );
+            const std::string message = ter_obj.get_string( "message" );
+            bool message_good = true;
+            if( ter_obj.has_bool( "message_good" ) ) {
+                message_good = ter_obj.get_bool( "message_good" );
+            }
+
+            while( target_array.has_more() ) {
+                const std::string valid_terrain = target_array.next_string();
+                ter_transform.emplace( ter_id( valid_terrain ), results );
+                if( !message.empty() ) {
+                    ter_message.emplace( ter_id( valid_terrain ), std::make_pair( message, message_good ) );
+                }
+            }
+
+            target_array = ter_obj.get_array( "valid_flags" );
+
+            while( target_array.has_more() ) {
+                const std::string valid_terrain = target_array.next_string();
+                ter_flag_transform.emplace( valid_terrain, results );
+                if( !message.empty() ) {
+                    ter_flag_message.emplace( valid_terrain, std::make_pair( message, message_good ) );
+                }
+            }
+        }
+    }
+
+    if( jo.has_member( "furniture" ) ) {
+        JsonArray obj_array = jo.get_array( "furniture" );
+        while( obj_array.has_more() ) {
+            JsonObject furn_obj = obj_array.next_object();
+            JsonArray target_array = furn_obj.get_array( "valid_terrain" );
+
+            weighted_int_list<furn_id> results;
+            load_furn_results( furn_obj, "result", results );
+            const std::string message = furn_obj.get_string( "message" );
+            bool message_good = true;
+            if( furn_obj.has_bool( "message_good" ) ) {
+                message_good = furn_obj.get_bool( "message_good" );
+            }
+
+            while( target_array.has_more() ) {
+                const std::string valid_terrain = target_array.next_string();
+                furn_transform.emplace( furn_id( valid_terrain ), results );
+                if( !message.empty() ) {
+                    furn_message.emplace( furn_id( valid_terrain ), std::make_pair( message, message_good ) );
+                }
+            }
+
+            target_array = furn_obj.get_array( "valid_flags" );
+
+            while( target_array.has_more() ) {
+                const std::string valid_terrain = target_array.next_string();
+                furn_flag_transform.emplace( valid_terrain, results );
+                if( !message.empty() ) {
+                    furn_flag_message.emplace( valid_terrain, std::make_pair( message, message_good ) );
+                }
+            }
+        }
+    }
+}
+
+ter_id ter_furn_transform::next_ter( const ter_id ter ) const
+{
+    const std::map<ter_id, weighted_int_list<ter_id>>::const_iterator result_iter =
```

Comment:
```suggestion
    const auto result_iter =
```
And than put the whole statement on one line.

Reply:
i've been led to believe that using auto is generally a bad thing, though i understand that iterators may be a special case. is this just an aesthetic choice?